app.rb

```diff
@@ -1,67 +1,44 @@
 require_relative './lib/players/player.rb'
 require_relative './lib/players/ai.rb'
 require_relative './lib/players/human.rb'
-require_relative './lib/players/player_script.rb'
-require_relative './lib/players/validator.rb'
+require_relative './lib/scripts/player_script.rb'
+require_relative './lib/validators/player_validator.rb'
 require_relative './lib/ui/user_interface.rb'
 require_relative './lib/ui/io.rb'
+require_relative './lib/scripts/game_script.rb'
+require_relative './lib/validators/setup_validator.rb'
+require_relative './lib/game_setup.rb'
 
-def initializer(size, token1, token2)
-  Struct.new('State', :stop_playing, :is_won, :restart)
-  state = Struct::State.new(false, false, false)
+def initializer(size)
+  Struct.new('GameProgress', :stop_playing, :is_won, :restart)
+  game_progress = Struct::GameProgress.new(false, false, false)
 
   board = Board.new(size, '-')
   checker = BoardChecker.new(board)
+
   io = IO.new(Readline, STDOUT)
-  user_interface = UserInterface.new(io, board, token1, token2)
-  validator = Validator.new(board)
-  human = Human.new(user_interface)
-  ai = AI.new(board, checker, token2, token1)
+  game_script = GameScript.new
+  user_interface = UserInterface.new(io, game_script, board)
+
+  player_validator = PlayerValidator.new(board)
   human_script = PlayerScript.new("\nPlease enter a number between 1 - 9: ", "\nYou picked spot ")
   ai_script = PlayerScript.new("\nThe computer is picking a spot...", "\nThe computer picks spot ")
-  player1 = Player.new(human_script, human, token1, user_interface, validator)
-  player2 = Player.new(ai_script, ai, token2, user_interface, validator)
-
-  trap_signals(user_interface)
-
-  until(state.stop_playing)
-    board.reset_board
-    user_interface.welcome
-    game = Game.new(state, board, checker, player1, player2)
-    game.play
-
-    unless(state.restart || state.stop_playing)
-      replay?(state, io)
-    end
 
-    state.restart = false
-  end
+  setup_validator = SetupValidator.new(game_script)
+  game_setup = GameSetup.new(user_interface, game_script, setup_validator)
+  game_options = game_setup.get_game_options
 
-  stop_playing(user_interface)
-end
-
-def trap_signals(user_interface)
-  signals = ['INT', 'TERM', 'QUIT','ABRT', 'TSTP']
-
-  signals.each do |signal|
-    trap(signal) do
-      stop_playing(user_interface)
-    end
-  end
-end
-
-def stop_playing(user_interface)
-  user_interface.exit_game
-  exit
-end
-
-def replay?(state, io)
-  replay = io.get_input("\n\nWould you like to replay [y/n]? ").downcase
-  unless(replay.to_sym == :yes || replay.to_sym == :y)
-    state.stop_playing = true
-  end
-end
+  human = Human.new(user_interface)
+  ai = AI.new(board, checker, game_options.token2, game_options.token1)
+  player1 = Player.new(human_script, human, game_options.token1, user_interface, player_validator)
+  player2 = Player.new(ai_script, ai, game_options.token2, user_interface, player_validator)
```

Comment:
so you're getting data back from game options, and applying that data to create players in the initialize function here, why not just have the game options return two players?